subgraphs/compound-forks/protocols/dforce/src/mapping.ts

```diff
@@ -603,33 +610,46 @@ export function handleStablecoinTransfer(event: StablecoinTransfer): void {
   let protocol = getOrCreateProtocol();
 
   let contract = stablecoin.bind(event.address);
+  let token = Token.load(tokenId);
+  if (token == null) {
+    token = new Token(tokenId);
+    token.name = contract.name();
+    token.symbol = contract.symbol();
+    token.decimals = contract.decimals();
+    token.save();
+  }
+
   let supply = contract.totalSupply();
   // since mintedTokens will be sorted by address, we need to make sure
   // mintedTokenSupplies is sorted in the same order
-  if (protocol.mintedTokens == null || protocol.mintedTokens!.length == 0) {
+  let mintedTokens = protocol.mintedTokens;
+  let mintedTokenSupplies = protocol.mintedTokenSupplies;
+  if (mintedTokens == null || mintedTokens.length == 0) {
     protocol.mintedTokens = [tokenId];
     protocol.mintedTokenSupplies = [supply];
   } else {
-    let tokenIndex = protocol.mintedTokens!.indexOf(tokenId);
-    if (tokenIndex > 0) {
+    let tokenIndex = mintedTokens.indexOf(tokenId);
+    if (tokenIndex >= 0) {
       // token already in protocol.mintedTokens
-      protocol.mintedTokenSupplies![tokenIndex] = supply;
+      mintedTokenSupplies![tokenIndex] = supply;
+      protocol.mintedTokenSupplies = mintedTokenSupplies;
     } else {
-      if (tokenId < protocol.mintedTokens![0]) {
+      if (tokenId < mintedTokens[0]) {
```

Comment:
Be careful with ordering here and checking that way (https://github.com/messari/subgraphs/blob/master/docs/Mapping.md#array-sorting-when-querying)

Reply:
The purpose of the whole chunk of code (L625-649) is to make sure the order of mintedToken and mintedTokenSupplies arrays are consistent.